### PR TITLE
Fix bug in Java OA constructor causing warning in finalizer

### PR DESCRIPTION
**Pull request by externl**
Ensure _state is set to StateDestroyed otherwise we sometimes get a warning from the finalizer. Saw this warning when running tests in a slower container.

```
- Config: ws,compress,ipv6,serialize,mx
testing load properties from UTF-8 path... ok
testing using Ice.Config with multiple config files... ok
testing configuration file escapes... ok
testing ice properties with set default values...ok
testing ice properties with unset default values...ok
testing that getting an unknown ice property throws an exception...ok
testing that setting an unknown ice property throws an exception...ok
testing that creating an object adapter with unknown properties throws an exception...ok
testing that creating a proxy with unknown properties throws an exception...ok
-! 10/22/24 18:37:52:772 warning: Finalizer: object adapter `FooOA' has not been deactivated
```

### CODE REVIEW

File: java/gradle/ice.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ spotless {
 }
 
 // Run spotlessApply before compiling Java code
-compileJava.dependsOn 'spotlessApply'
+// compileJava.dependsOn 'spotlessApply'
```
Comment (pepone) [inline java/gradle/ice.gradle:85]: Why is this commented out?

**Review by pepone — APPROVED** [review]
Looks good, but we should remove Java finalize which was deprecated in Java 9. For a future PR.

**Review by bernardnormier — APPROVED** [review]

File: java/gradle/ice.gradle (hunk 1 above)
Comment (externl) [reply java/gradle/ice.gradle:85]: We shouldn't be doing this. Otherwise we'll never know if the formatting is bad.

Comment (pepone) [reply java/gradle/ice.gradle:85]: then let's remove it.

**Pull request merged**